src/main/java/seedu/realodex/logic/parser/PrefixChecker.java

```diff
@@ -54,30 +61,62 @@ public boolean checkEmptyPreamble() {
     }
 
     /**
-     * Verifies that no duplicate prefixes are present for the provided prefixes.
-     * A duplicate prefix is defined as the same prefix being used multiple times with different values.
+     * Checks for duplicate prefixes in the given array of prefixes, with an exception for tag prefixes
+     * that is allowed to appear at most twice. Throws a {@code ParseException} if duplicates are found.
      *
-     * @param prefixes The prefixes to check for duplicates.
-     * @throws ParseException if duplicate prefixes are found.
+     * @param prefixes An array of {@code Prefix} objects to be checked for duplicates.
+     * @throws ParseException If duplicates are found, except for the allowed exceptions.
      */
-    public void checkNoDuplicatePrefix(Prefix...prefixes) throws ParseException {
-        Prefix[] duplicatedPrefixes = Stream.of(prefixes).distinct()
-                .filter(prefix -> argumentMultimap.containsPrefix(prefix)
-                        && argumentMultimap.getAllValues(prefix).size() > 1)
+    public void checkNoDuplicatePrefix(Prefix... prefixes) throws ParseException {
+        assert(prefixes != null);
+        Prefix[] duplicatedPrefixes = Stream.of(prefixes)
+                .filter(this::isDuplicatePrefix)
+                .distinct()
                 .toArray(Prefix[]::new);
 
         if (duplicatedPrefixes.length > 0) {
             throw new ParseException(Messages.getErrorMessageForDuplicatePrefixes(duplicatedPrefixes));
         }
     }
 
+    /**
+     * Determines whether a given prefix is considered a duplicate within the context of an {@code ArgumentMultimap}.
+     * A special case is made for special prefixes where at most 1 duplicate is allowed.
+     *
+     * @param prefix The {@link Prefix} to check for duplication.
+     * @return {@code true} if the prefix is considered a duplicate (considering the allowed exceptions),
+     *         {@code false} otherwise.
+     */
+    public boolean isDuplicatePrefix(Prefix prefix) {
+        assert(prefix != null);
+        if (isSpecialCasePrefix(prefix)) {
+            return argumentMultimap.containsPrefix(prefix) && argumentMultimap.getAllValues(prefix).size() > 2;
+        } else {
+            return argumentMultimap.containsPrefix(prefix) && argumentMultimap.getAllValues(prefix).size() > 1;
+        }
+    }
+
+    /**
+     * Identifies if the given prefix is a special case that is subject to different duplication rules.
+     * For example, tag prefixes are allowed to appear more than once but at most twice
+     * without being considered duplicates.
+     *
+     * @param prefix The {@link Prefix} to be checked.
+     * @return {@code true} if the prefix is a special case, {@code false} otherwise.
+     */
+    public boolean isSpecialCasePrefix(Prefix prefix) {
+        assert(prefix != null);
```

Comment:
Actually I used it because I only use this within the PrefixChecker class, and everytime I call it, I have already checked that the Prefix is not null. 

Hence, as a programmer, I am _assuming_ that the input here will never be `null` is that okay?